lib/spack/spack/installer.py

```diff
@@ -2019,58 +2101,29 @@ def install(self) -> None:
         fail_fast_err = "Terminating after first install failure"
         single_requested_spec = len(self.build_requests) == 1
         failed_build_requests = []
-
         install_status = InstallStatus(len(self.build_pq))
+        active_tasks: List[Task] = []
 
         # Only enable the terminal status line when we're in a tty without debug info
         # enabled, so that the output does not get cluttered.
         term_status = TermStatusLine(
             enabled=sys.stdout.isatty() and tty.msg_enabled() and not tty.is_debug()
         )
 
-        while self.build_pq:
-            task = self._pop_task()
-            if task is None:
-                continue
-
-            install_args = task.request.install_args
-            keep_prefix = install_args.get("keep_prefix")
-
+        def start_task(task) -> Optional[Task]:
```

Comment:
Since you don't use the return value, you can just make this return nothing
```suggestion
        def start_task(task) -> None:
```